Add pull request template (#1827)
jamulussoftware/jamulus: .github/pull_request_template.md

> <!-- Thank you for working on Jamulus and opening a Pull Request! Please fill the following to make maintenance and development process visible -->

Reviewer: Not exactly sure I understand what this is meant to say. Maybe something like this?

```suggestion
<!-- Thank you for working on Jamulus and opening a Pull Request! Please fill the following to make the review process straight foward -->
```

Author: It's not only about the review process, but also about the process of development (if there's any on the PR)

**Short description of changes**

<!-- A short description of your changes which might go into the change log -->

**Context: Fixes an issue?**

<!-- If this fixes an issue, please write Fixes: <issue number here>; if not, please give your PR a context. -->

**Does this change need documentation? What needs to be documented and how?**

<!-- Most new features should be documented on the website: https://github.com/jamulussoftware/jamuluswebsite/ If you have a proposal what to document, feel free to open a draft PR on the website repo -->

> **Status of this pull request**

Reviewer: The other places use upper case:

```suggestion
**Status of this Pull Request**
```

<!-- This might be edited by maintainers. -->
<!-- Proof of concept (not to be merged soon); Working implementation; ... -->

**What is missing until this pull request can be merged?**
<!-- Does it still need more testing; ... -->

> ## Checklist

Reviewer: ```suggestion
## Checklist
<!-- Please tick the check boxes when done by replacing the space by an x, e.g. [x]. -->
```

> - [ ] I verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)

Reviewer: ```suggestion
- [ ] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
```

> - [ ] I tested my code and it does what I want

Reviewer: Sounds a bit vague. Should we make it more specific? Some ideas:
- I have documented the platforms I've tested this PR on.
- I have documented how this PR can be tested and its correctness can be verified.
?

Reviewer 2: Agreed.

Jamulus does actually have a test suite...  Ideally, we'd be maintaining it.

For all parts of the system not covered (most of the program), we'd still write out the spec / test guideance notes used, as part of the discussion or PR itself.

But still, it's not going to stop code getting reviewed and accepted.

- [ ] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) <!-- You can also check if your code passes clang-format -->

> - [ ] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. <!-- GitHub doesn't run these checks for new contributors automatically. -->

Reviewer: ```suggestion
- [ ] I've waited some time after this Pull Request was opened and all GitHub checks completed without errors. <!-- GitHub doesn't run these checks for new contributors automatically. If nobody main developer has approved the checks after a few days, feel free to post a quick ping. -->
```

Author: > If nobody main developer has approved the checks after a few days, feel free to post a quick ping

This might add pressure. Pinging should not be too common in my opinion.

<!-- Most new features should be documented on the website: https://github.com/jamulussoftware/jamuluswebsite/ If you have a proposal what to document, feel free to open a draft PR on the website repo -->

**Status of this Pull Request**
<!-- This might be edited by maintainers. -->
<!-- Proof of concept (not to be merged soon); Working implementation; ... -->

**What is missing until this pull request can be merged?**
<!-- Does it still need more testing; ... -->

## Checklist
<!-- Please tick the check boxes when done by replacing the space by an x, e.g. [x]. -->
- [ ] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [ ] I tested my code and it does what I want
- [ ] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) <!-- You can also check if your code passes clang-format -->
- [ ] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. <!-- GitHub doesn't run these checks for new contributors automatically. -->

> - [ ] I've filled all the content above

Author: @dcorson-ticino-com would this wording be ok?
